tests/AccountTest.php

```diff
@@ -2,469 +2,156 @@
     {
-        $this->mockRequest(
-            'POST',
-            '/v1/accounts',
-            array('managed' => 'true'),
-            $this->managedAccountResponse('acct_ABC')
+        $this->expectsRequest(
+            'get',
+            '/v1/accounts/' . TEST_RESOURCE_ID
         );
-        $account = Account::create(array(
-            'managed' => true
-        ));
-        $this->assertSame($account->id, 'acct_ABC');
+        $resource = Account::retrieve(TEST_RESOURCE_ID);
+        $this->assertSame("Stripe\\Account", get_class($resource));
     }
 
-    public function testDelete()
+    public function testIsRetrievableWithoutId()
     {
-        $account = self::createTestAccount();
-
-        $this->mockRequest(
-            'DELETE',
-            '/v1/accounts/' . $account->id,
-            array(),
-            $this->deletedAccountResponse('acct_ABC')
+        $this->expectsRequest(
+            'get',
+            '/v1/account'
         );
-        $deleted = $account->delete();
-        $this->assertSame($deleted->id, $account->id);
-        $this->assertTrue($deleted->deleted);
+        $resource = Account::retrieve();
+        $this->assertSame("Stripe\\Account", get_class($resource));
     }
 
-    public function testReject()
+    public function testIsCreatable()
     {
-        $account = self::createTestAccount();
-
-        $this->mockRequest(
-            'POST',
-            '/v1/accounts/' . $account->id . '/reject',
-            array('reason' => 'fraud'),
-            $this->deletedAccountResponse('acct_ABC')
+        $this->expectsRequest(
+            'post',
+            '/v1/accounts'
         );
-        $rejected = $account->reject(array('reason' => 'fraud'));
-        $this->assertSame($rejected->id, $account->id);
+        $resource = Account::create(array("type" => "custom"));
+        $this->assertSame("Stripe\\Account", get_class($resource));
     }
 
-    public function testSaveLegalEntity()
+    public function testIsSaveable()
     {
-        $response = $this->managedAccountResponse('acct_ABC');
-        $this->mockRequest('POST', '/v1/accounts', array('managed' => 'true'), $response);
-
-        $response['legal_entity']['first_name'] = 'Bob';
-        $this->mockRequest(
-            'POST',
-            '/v1/accounts/acct_ABC',
-            array('legal_entity' => array('first_name' => 'Bob')),
-            $response
+        $resource = Account::retrieve(TEST_RESOURCE_ID);
+        $resource->metadata["key"] = "value";
+        $this->expectsRequest(
+            'post',
+            '/v1/accounts/' . TEST_RESOURCE_ID
         );
-
-        $account = Account::create(array('managed' => true));
-        $account->legal_entity->first_name = 'Bob';
-        $account->save();
-
-        $this->assertSame('Bob', $account->legal_entity->first_name);
+        $resource->save();
+        $this->assertSame("Stripe\\Account", get_class($resource));
     }
 
-    public function testUpdateLegalEntity()
```

Comment:
@ob-stripe I removed those tests but I am worried that we're not testing this properly anymore. Should this live here or in a separate test that properly handles sub-fields/nested resources updates?

Reply:
Good question. I suppose since this is all handled in `StripeObject` directly, we should test this in `StripeObjectTest`.